static/app/views/settings/organizationAuthTokens/authTokenDetails.tsx

```diff
@@ -103,25 +104,34 @@ function AuthTokenDetailsForm({
       );
 
       // Update get list query
-      setApiQueryData(
-        queryClient,
-        makeFetchOrgAuthTokensForOrgQueryKey({orgSlug: organization.slug}),
-        (oldData: OrgAuthToken[] | undefined) => {
-          if (!Array.isArray(oldData)) {
-            return oldData;
-          }
-
-          const existingToken = oldData.find(oldToken => oldToken.id === token.id);
+      if (
+        getApiQueryData(
```

Comment:
m: Can we split up this `useMutation` like so https://tkdodo.eu/blog/mastering-mutations-in-react-query#some-callbacks-might-not-fire?

This should mean we can remove `window.setTimeout` call since `onSuccess` in `mutate` call shouldn't fire on redirect.

Reply:
Actually hmm, not sure if that will help here. In general I think the warning being emitted is a false positive - see https://github.com/reactwg/react-18/discussions/82

Reply:
Luckily we already ran into this https://github.com/getsentry/sentry/pull/41546!

Reply:
Does this still produce a warning in the tests? I would like to keep tests warning free

Reply:
I didn't notice any warnings/errors in tests at all, just noticed this while running the app locally!